Cut(eos_designs)!: Remove the deprecated keys from defs_p2p_links schema (#5921)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

| `aaa_settings.accounting.commands[].default[].group` | `aaa_settings.accounting.commands[].default[].methods[].group` |
| `aaa_settings.accounting.commands[].default[].logging` | `aaa_settings.accounting.commands[].default[].methods[].method` |
| `avd_6_behaviors` | - |

> | `core_interfaces.p2p_links[].structured_config` | `core_interfaces.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |

Reviewer: ```suggestion
| `core_interfaces.p2p_links[].structured_config` | `core_interfaces.p2p_links.ethernet_structured_config` or `core_interfaces.p2p_links.port_channel_structured_config` |
```

| `core_interfaces.p2p_links[].underlay_multicast` | `core_interfaces.p2p_links.multicast_pim_sm` |

> | `core_interfaces.p2p_links_profiles[].structured_config` | `core_interfaces.p2p_links_profiles.ethernet_structured_config` or `l3_edge.p2p_links_profiles.port_channel_structured_config` |

Reviewer: ```suggestion
| `core_interfaces.p2p_links_profiles[].structured_config` | `core_interfaces.p2p_links_profiles.ethernet_structured_config` or `core_interfaces.p2p_links_profiles.port_channel_structured_config` |
```

| `aaa_settings.accounting.commands[].default[].logging` | `aaa_settings.accounting.commands[].default[].methods[].method` |
| `avd_6_behaviors` | - |
| `core_interfaces.p2p_links[].underlay_multicast` | `core_interfaces.p2p_links.multicast_pim_sm` |

> | `core_interfaces.p2p_links[].structured_config` | `core_interfaces.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |

Reviewer 2: Add for `core_interfaces.p2p_links_profiles.[].structured_config` and `l3_edge.p2p_links_profiles.[].structured_config` as well

| `aaa_settings.accounting.commands[].default[].logging` | `aaa_settings.accounting.commands[].default[].methods[].method` |
| `avd_6_behaviors` | - |
| `core_interfaces.p2p_links[].underlay_multicast` | `core_interfaces.p2p_links.multicast_pim_sm` |

> | `core_interfaces.p2p_links[].structured_config` | `core_interfaces.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |

Reviewer: ```suggestion
| `core_interfaces.p2p_links[].structured_config` | `core_interfaces.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |
| `core_interfaces.p2p_links[].underlay_multicast` | `core_interfaces.p2p_links.multicast_pim_sm` |
```

| `core_interfaces.p2p_links_profiles[].underlay_multicast` | `core_interfaces.p2p_links_profiles.multicast_pim_sm` |

> | `core_interfaces.p2p_links_profiles[].structured_config` | `core_interfaces.p2p_links_profiles.ethernet_structured_config` or `l3_edge.p2p_links_profiles.port_channel_structured_config` |

Reviewer: ```suggestion
| `core_interfaces.p2p_links_profiles[].structured_config` | `core_interfaces.p2p_links_profiles.ethernet_structured_config` or `l3_edge.p2p_links_profiles.port_channel_structured_config` |
| `core_interfaces.p2p_links_profiles[].underlay_multicast` | `core_interfaces.p2p_links_profiles.multicast_pim_sm` |
```

| `cvp_ingestauth_key` | - |
| `cvp_instance_ips` | `cv_settings` |
| `cvp_token_file` | `cv_settings` |
| `design` | - |
| `flow_tracking_settings.trackers.exporters.collector` | `flow_tracking_settings.trackers.exporters.collectors` |
| `l3_edge.p2p_links[].underlay_multicast` | `l3_edge.p2p_links.multicast_pim_sm` |
| `l3_edge.p2p_links[].structured_config` | `l3_edge.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |
| `l3_edge.p2p_links_profiles[].underlay_multicast` | `l3_edge.p2p_links_profiles.multicast_pim_sm` |

> | `l3_edge.p2p_links_profiles[].structured_config` | `l3_edge.p2p_links_profiles.ethernet_structured_config` or `l3_edge.p2p_links_profiles.port_channel_structured_config` |

Reviewer: ```suggestion
| `l3_edge.p2p_links[].structured_config` | `l3_edge.p2p_links.ethernet_structured_config` or `l3_edge.p2p_links.port_channel_structured_config` |
| `l3_edge.p2p_links[].underlay_multicast` | `l3_edge.p2p_links.multicast_pim_sm` |
| `l3_edge.p2p_links_profiles[].structured_config` | `l3_edge.p2p_links_profiles.ethernet_structured_config` or `l3_edge.p2p_links_profiles.port_channel_structured_config` |
| `l3_edge.p2p_links_profiles[].underlay_multicast` | `l3_edge.p2p_links_profiles.multicast_pim_sm` |
```

| `<node_type_keys.key>.node_groups.[].l3_port_channels.[].member_interfaces.[].speed` | Added valid_values |
| `<node_type_keys.key>.nodes.[].l3_port_channels.[].member_interfaces.[].speed` | Added valid_values |
| `p2p_links[].speed` | Added valid_values |
| `p2p_links[].ip` | Was too permissive before and broke with more than 2 IPs. Now limited to exactly 2 entries. |
| `p2p_links[].nodes` | Was too permissive before and broke with more than 2 nodes. Now limited to exactly 2 entries. |
| `p2p_links[].interfaces` | Was too permissive before and broke with more than 2 interfaces. Now limited to exactly 2 entries. |

> | `p2p_links[].description` | Was too permissive before and broke with more than 2 descriptions. Now limited to exactly 2 entries. |

Reviewer: Hmm let's keep it simple here:

min_length: 2, max_length: 2

and add an extra paragraph entry for all the changes - something like this:

### p2p_links now accepts a minimum and maximum of two entries

In AVD 5.0.0, `p2p_links` list keys `nodes`, `ip`, `interface` and `description` did not have any limitation. This was sometimes working but could lead to undefined behavior as all the code was written to work for exactly two nodes, in particular when figuring out the peer device.

Starting AVD 6.0.0, the length to all these lists has been restricted to exactly 2 items (min_length: 2 and max_length: 2).

Author: updated. Thanks!!

+     underlay_ethernet_interfaces: 'custom_templates/interface_descriptions/underlay/ethernet-interfaces.j2'
```

### p2p_links now accepts a minimum and maximum of two entries

> In AVD 5.0.0, p2p_links list keys nodes, ip, interfaces and descriptions did not have any limitation. This was sometimes working but could lead to undefined behavior as all the code was written to work for exactly two nodes, in particular when figuring out the peer device.

Reviewer: ```suggestion
In AVD 5.0.0, p2p_links list keys `nodes`, `ip`, `interfaces` and `descriptions` did not have any limitation in size. This was sometimes working but could lead to undefined behavior as all the code was written to work for exactly `2` nodes, in particular when figuring out the peer device.
```

> Starting AVD 6.0.0, the length to all these lists has been restricted to exactly 2 items (min_length: 2 and max_length: 2).

Reviewer: ```suggestion
Starting AVD 6.0.0, the length to all these lists have been restricted to exactly 2 items (`min_length: 2` and `max_length: 2`).
```
